src/render/metal/SDL_render_metal.m

```diff
@@ -279,26 +281,35 @@ - (void)dealloc
 
     switch (cache->vertexFunction) {
         case SDL_METAL_VERTEX_SOLID:
-            /* position (float2) */
-            vertdesc.layouts[0].stride = sizeof(float) * 2;
+            /* position (float2), color (float4) */
```

Comment:
colors are unsigned normalized 8 bit values in the higher level API, so you don't need the uint8 -> float conversion in backends like Metal where it's easy to pass the same type all the way through to the GPU.

Reply:
Thanks for the info, I think gles2 back-end can do the same with (glVertexAttribPointer with 'normalized' param).
I am not so familiar with metal to change this quickly, I'll do it probably when I have some spare time. (no need to change the shader, right ?)

Reply:
Right. It's not a big deal either way, it just might simplify some code (or it might add more complexity elsewhere, I'm not sure).

Reply:
I've just modified this way ..